### PR TITLE
Switch FAST networking stages to network policies for Google domains

### LINKED ISSUE #1292: Migrate FAST to DNS response policies
Update FAST networking stages to use the new `dns-response-policy` module for Google endpoints.

### PR DESCRIPTION
**Pull request by ludoo**
This fixes #1292 by removing all private zones supporting Google domains for APIs, and reimplementing them via a single response policy for added flexibility.

image

Updated to implement a factory in the DNS resource policy module, and to use corresponding factory files in FAST networking stages.

### CODE REVIEW

File: fast/stages/2-networking-a-peering/dns-landing.tf [hunk 1]
```diff
@@ -50,82 +78,43 @@ module "gcp-example-dns-private-zone" {
...
+        dns_name = "restricted.googleapis.com."
+        local_data = { A = { rrdatas = [
+          "199.36.153.4", "199.36.153.5", "199.36.153.6", "199.36.153.7"
+        ] } }
+      }
+    },
+    {
+      for k, v in local.googleapis_domains : k => {
+        dns_name = v
+        local_data = { CNAME = { rrdatas = [
```
Comment (drebes) [inline fast/stages/2-networking-a-peering/dns-landing.tf:114]: Will the naked domains (like "gcr.io") work with CNAMEs? I don't think so. (They usually require explicit pointing to the IPs).

Comment (ludoo) [reply fast/stages/2-networking-a-peering/dns-landing.tf:114]: let me try... they seem to work

```
ext_ludomagno_google_com@bastion:~$ dig pkg.dev +short
private.googleapis.com.
199.36.153.8
199.36.153.9
199.36.153.10
199.36.153.11
```

File: fast/stages/2-networking-b-vpn/dns-landing.tf [hunk 2]
```diff
@@ -50,82 +78,43 @@ module "gcp-example-dns-private-zone" {
...
-
-module "gcrio-private-zone" {
-  source          = "../../../modules/dns"
-  project_id      = module.landing-project.project_id
-  type            = "private"
-  name            = "gcr-io"
-  domain          = "gcr.io."
-  client_networks = [module.landing-vpc.self_link]
-  recordsets = {
-    "A gcr.io." = { ttl = 300, records = [
```
Comment (drebes) [inline fast/stages/2-networking-b-vpn/dns-landing.tf:81]: I didn't find this on the policies. (See my other comment about naked domains)

Comment (ludoo) [reply fast/stages/2-networking-b-vpn/dns-landing.tf:81]: what didn't you find?

**Review by juliocc — COMMENTED** [review]
beautiful 

Can we include the other domains mentioned [here](https://cloud.google.com/vpc/docs/configure-private-google-access#domain-options

**Review by juliocc — APPROVED** [review]

**Pull request merged**